code/game/turfs/simulated/wall_icon.dm

```diff
@@ -66,54 +66,54 @@
 	if(!damage_overlays[1]) //list hasn't been populated; note that it is always of fixed length, so we must check for membership.
 		generate_overlays()
 
-	overlays.Cut()
 
+	var/new_overlays
 	var/material_icon_base = get_wall_state()
 	var/image/I
 	var/base_color = paint_color ? paint_color : material.icon_colour
 	if(!density)
 		I = image(icon, "[material_icon_base]fwall_open")
 		I.color = base_color
-		overlays += I
+		LAZYADD(new_overlays, I)
 		return
```

Comment:
need to watch for early returns with this setup; this returns without setting overlays to `new_overlays`.

Reply:
Ah shoot so it does.